### PR TITLE
Add compact-shard sub-command to influx-tools tool

### PR DESCRIPTION
**Pull request by stuartcarnie**

### CODE REVIEW

**Comment by hercules-influx** [comment]
During a run of [megacheck](https://github.com/dominikh/go-tools) the following issues were discovered:
``````

**Comment by stuartcarnie** [comment]
I use the store directly as is it handles merging the levels and WAL via the cursors. There isn't an easy way to do this without replicating a lot of code. To reduce start up time and memory overhead, this command uses the `EngineOptions` predicates (added for the Export tool) to open open the specified shard, so it should have much lower memory requirements and spin up storage much faster. See [this section][diff] of the diff.

[diff]: https://github.com/influxdata/influxdb/pull/9963/files#diff-1a1590d37f0be8c92d200e549d5771edR297

File: cmd/influx-tools/compact/command.go [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+			return nil
+		}
+	}
+
+	gen, seq := 1, 4
+	if len(files.tsm) > 0 {
+		sort.Strings(files.tsm)
+		gen, _, err = tsm1.DefaultParseFileName(files.tsm[len(files.tsm)-1])
+		if err != nil {
+			return cmd.fmtError(fmt.Sprintf("failed to parse tsm file %q: %v", files.tsm[len(files.tsm)-1], err))
```
[the remark this answers is not in the document]
Comment (stuartcarnie) [reply cmd/influx-tools/compact/command.go]: I'll do that 👍

File: cmd/influx-tools/compact/command.go [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+	}
+
+	sw := shard.NewWriter(cmd.shardID, dataPath, shard.Temporary(), shard.Generation(gen), shard.Sequence(seq))
+
+	var countSeries, countValues int
+
+	values := make(tsm1.Values, 1000) // max block size
+	var keyFieldSeparatorBytes = []byte("#!~#")
+
+	makeKey := func(name []byte, tags models.Tags, field []byte) []byte {
```
[the remark this answers is not in the document]
Comment (stuartcarnie) [reply cmd/influx-tools/compact/command.go]: Plagiarized. Added an extra byte for the delimiter between `name` and `tags`

**Pull request merged**